nifi-nar-bundles/nifi-cassandra-bundle/nifi-cassandra-processors/src/main/java/org/apache/nifi/processors/cassandra/PutCassandraRecord.java

```diff
@@ -193,6 +303,81 @@ public void onTrigger(ProcessContext context, ProcessSession session) throws Pro
 
     }
 
+    private Statement generateUpdate(String cassandraTable, RecordSchema schema, String updateKeys, String updateMethod, Map<String, Object> recordContentMap) {
+        Update updateQuery;
+
+        // Split up the update key names separated by a comma, should not be empty
+        final Set<String> updateKeyNames;
+        updateKeyNames = Arrays.stream(updateKeys.split(","))
+                .map(String::trim)
+                .filter(StringUtils::isNotEmpty)
+                .collect(Collectors.toSet());
+        if (updateKeyNames.isEmpty()) {
+            throw new IllegalArgumentException("No Update Keys were specified");
+        }
+
+        // Verify if all update keys are present in the record
+        for (String updateKey : updateKeyNames) {
+            if (!schema.getFieldNames().contains(updateKey)) {
+                throw new IllegalArgumentException("Update key '" + updateKey + "' is not present in the record schema");
+            }
+        }
+
+        // Prepare keyspace/table names
+        if (cassandraTable.contains(".")) {
+            String[] keyspaceAndTable = cassandraTable.split("\\.");
+            updateQuery = QueryBuilder.update(keyspaceAndTable[0], keyspaceAndTable[1]);
+        } else {
+            updateQuery = QueryBuilder.update(cassandraTable);
+        }
+
+        // Loop through the field names, setting those that are not in the update key set, and using those
+        // in the update key set as conditions.
+        for (String fieldName : schema.getFieldNames()) {
+            Object fieldValue = recordContentMap.get(fieldName);
+
+            if (updateKeyNames.contains(fieldName)) {
+                updateQuery.where(QueryBuilder.eq(fieldName, fieldValue));
+            } else {
+                Assignment assignment;
+                if (SET_TYPE.getValue().equalsIgnoreCase(updateMethod)) {
+                    assignment = QueryBuilder.set(fieldName, fieldValue);
+                } else {
+                    // Check if the fieldValue is of type long, as this is the only type that is can be used,
+                    // to increment or decrement.
+                    if (!(fieldValue instanceof Long)) {
+                        throw new IllegalArgumentException("Field '" + fieldName + "' is not of type Long, and cannot be used" +
+                                " to increment or decrement.");
+                    }
+
+                    if (INCR_TYPE.getValue().equalsIgnoreCase(updateMethod)) {
+                        assignment = QueryBuilder.incr(fieldName, (Long)fieldValue);
+                    } else if (DECR_TYPE.getValue().equalsIgnoreCase(updateMethod)) {
+                        assignment = QueryBuilder.decr(fieldName, (Long)fieldValue);
+                    } else {
+                        throw new IllegalArgumentException("Update Method '" + updateMethod + "' is not valid.");
```

Comment:
Agreed.

Reply:
Is it intentional you mean?

If for example we want to set string fields, but have an error in the update method (let's say it has a type in it like 'SED' instead of 'SET'), the encountered error message "Field is not of type Long..." would be very misleading.

In general the validity of the update method _itself_ is a higher level issue than the validity of the _parameters_ of the update method and as such might be better to check that first.

Reply:
Ah, no I meant that I see your point and will push a commit to that effect :-).